ergo-merkle-tree/src/batchmerkleproof.rs

```diff
@@ -0,0 +1,210 @@
+use crate::LevelNode;
+use crate::NodeSide;
+use crate::{concatenate_hashes, prefixed_hash, prefixed_hash2};
+use crate::{HASH_SIZE, INTERNAL_PREFIX};
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "json", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+    feature = "json",
+    serde(into = "crate::json::BatchMerkleProofJson"),
+    serde(try_from = "crate::json::BatchMerkleProofJson")
+)]
+#[cfg_attr(feature = "arbitrary", derive(proptest_derive::Arbitrary))]
+/// Compact Merkle multiproof. Can be created using [`crate::MerkleTree::proof_by_indices`]
+pub struct BatchMerkleProof {
+    pub(crate) indices: Vec<(usize, [u8; 32])>,
```

Comment:
Hmm, I do think it will be a bit tricky, since I moved ExtensionCandidate to ergo-chain-types, and that depends on ergo_merkle_tree, so that'd induce a circular dependency. One way would be to just move ExtensionCandidate::proof_for_interlink_vector to [NipopowAlgos](https://github.com/ergoplatform/ergo/blob/master/src/main/scala/org/ergoplatform/modifiers/history/popow/NipopowAlgos.scala#L296), however I think the easier route will just be to re-import digest32::Digest into ergo_merkle_tree instead of importing it from ergo-chain-types. I think the latter option is preferable

Reply:
> Hmm, I do think it will be a bit tricky, since I moved ExtensionCandidate to ergo-chain-types, and that depends on ergo_merkle_tree, so that'd induce a circular dependency. I think I will move ExtensionCandidate::proof_for_interlink_vector to [NipopowAlgos](https://github.com/ergoplatform/ergo/blob/master/src/main/scala/org/ergoplatform/modifiers/history/popow/NipopowAlgos.scala#L296).

Good call! Also, another option is to extract Digest32 and stuff it into a new `ergo-primitive-chain-types` crate but I'd rather use it as a last resort since our dependency tree already quite complex.

Reply:
If you decide to leave `[u8; HASH_SIZE]` as is please make a type alias for it.

Reply:
All ergo_merkle_tree types are using Digest32 now.